src/containers/CreatePipelineRun/CreatePipelineRun.js

```diff
@@ -318,15 +318,24 @@ class CreatePipelineRun extends React.Component {
   };
 
   initialState = () => {
-    const { namespace, pipelineRef } = this.props;
+    const { namespace } = this.props;
+    let { pipelineRef } = this.props;
     const pipelineInfo = parsePipelineInfo(pipelineRef, namespace);
+    if (
```

Comment:
This whole block here doesn't look right... are you sure there's not a neater way of doing things?

Surely pipelineRef is nothing in the case of the Pipeline no longer exists, so why do we need to do the undefined checks as well

Reply:
I wasnt sure if getting a pipelineError is not definitive of the pipeline not existing, I tihnk you can get these for other reasons 
I wanted to add an extra check(s) to make sure it wasnt just an error, I could change it to just check if there is an error then set it to empty?

Reply:
But if there's an error it doesn't matter what paramSpecs and resourceSpecs are surely, so you could just omit those - did you see any value in adding them?

Reply:
I have taken them out and left it as just the error, I dont think they would add any value